File: cometchat_calls_sdk/ios/Classes/Helper/PluginConstants.swift
import Foundation

class PluginConstants {
    
    @objc public enum Id:Int {
        case CALLING_NATIVE_VIEW_ID
        case METHOD_CHANNEL_NAME
        case EVENT_CHANNEL_NAME
        
        var value: String {
            switch self {
            case .CALLING_NATIVE_VIEW_ID:
                return "cometchatcallsNativeViewiOS"
            case .METHOD_CHANNEL_NAME:
                return "cometchatcalls_plugin"
            case .EVENT_CHANNEL_NAME:
                return "cometchatcalls_events_stream"
            }
        }
    }
    
    @objc public enum Method:Int {
        case INIT_COMETCHAT_CALLS
        case GENERATE_TOKEN
        case START_SESSION
        case END_SESSION
        case SWITCH_CAMERA
        case MUTE_AUDIO
        case PAUSE_VIDEO
        case SET_AUDIO_MODE
        case ENTER_PIP_MODE
        case EXIT_PIP_MODE
        case SWITCH_TO_VIDEO_CALL
        case START_RECORDING
        case STOP_RECORDING
        
        var value: String {
            switch self {
                
            case .INIT_COMETCHAT_CALLS:
                return "initCometChatCalls"
            case .GENERATE_TOKEN:
                return "generateToken"
            case .START_SESSION:
                return "startSession"
            case .END_SESSION:
                return "endSession"
            case .SWITCH_CAMERA:
                return "switchCamera"
            case .MUTE_AUDIO:
                return "muteAudio"
            case .PAUSE_VIDEO:
                return "pauseVideo"
            case .SET_AUDIO_MODE:
                return "setAudioMode"
            case .ENTER_PIP_MODE:
                return "enterPIPMode"
            case .EXIT_PIP_MODE:
                return "exitPIPMode"
            case .SWITCH_TO_VIDEO_CALL:
                return "switchToVideoCall"
            case .START_RECORDING:
                return "startRecording"
            case .STOP_RECORDING:
                return "stopRecording"
            }
        }
    }
    
    @objc public enum ErrorMessage:Int {
        case GENERATED_TOKEN_NULL
        case CALL_TOKEN_NULL
        case EVENT_STREAM_SINK_NULL
        case SETTINGS_METHOD_CALL_ERROR
        case CALLS_SDK_NOT_INITIALIZED
        case ARGUMENT_IS_NULL
        
        var value: String {
            switch self {
                
            case .GENERATED_TOKEN_NULL:
                return "Generated token is null"
            case .CALL_TOKEN_NULL:
                return "Call token is null"
            case .EVENT_STREAM_SINK_NULL:
                return "Event stream sink is null"
            case .SETTINGS_METHOD_CALL_ERROR:
                return "Unable to process your request"
            case .CALLS_SDK_NOT_INITIALIZED:
                return "Please initialize the CometChatCalls by calling CometChatCalls.init() method"
            case .ARGUMENT_IS_NULL:
                return "Passed argument is null"
            }
        }
    }
    
    @objc public enum SuccessMessage:Int {
        case INIT
        case SUCCESS
        
        var value: String {
            switch self {
                
            case .INIT:
                return "CometChat SDK initialized successfully"
            case .SUCCESS:
                return "success"
            }
        }
    }
    
    
    @objc public enum Code:Int {
        case SUCCESS
        case ERROR
        
        var value:String {
            switch self {
            case .ERROR:
                return "ERROR"
            case .SUCCESS:
                return "SUCCESS"
            }
        }
    }
    
    @objc public enum Position:Int {
        
        case TOP_LEFT
        case TOP_RIGHT
        case BOTTOM_LEFT
        case BOTTOM_RIGHT
        
        var value:String {
            
            switch self {
                
            case .TOP_LEFT:
                return "top-left"
            case .TOP_RIGHT:
                return "top-right";
            case .BOTTOM_LEFT:
                return "bottom-left";
            case .BOTTOM_RIGHT:
                return "bottom-right";
            }
        }
    }
    
    @objc public enum AspectRatio:Int {
        
        case DEFAULT
        case CONTAIN
        case COVER
        
        var value:String {
            
            switch self {
                
            case .DEFAULT:
                fallthrough
            case .CONTAIN:
                return "contain";
            case .COVER:
                return "cover";
            }
        }
    }
}

File: cometchat_calls_sdk/android/src/main/kotlin/com/cometchatcalls/cometchatcalls_plugin/helper/PluginConstants.kt
package com.cometchatcalls.cometchatcalls_plugin.helper

/**
 * Created by Rohit Giri on 15/05/23.
 */
object PluginConstants {
    object Id {
        const val METHOD_CHANNEL_NAME = "cometchatcalls_plugin"
        const val EVENT_CHANNEL_NAME = "cometchatcalls_events_stream"
        const val CALLING_NATIVE_VIEW_ID = "cometchatcallsNativeViewAndroid"
    }

    object CometchatCallsInitArgumentsKeys {
        const val APP_ID = "appID"
        const val REGION = "region"
        const val HOST = "host"
    }

    object CometchatCallsGenerateTokenArgumentsKeys{
        const val SESSION_ID = "sessionId"
        const val USER_AUTH_TOKEN = "userAuthToken"
    }

    object CometchatCallsStartSessionArgumentsKeys{
        const val CALL_TOKEN = "callToken"
    }

    object Method{
        const val INIT_COMETCHAT_CALLS = "initCometChatCalls"
        const val GENERATE_TOKEN = "generateToken"
        const val START_SESSION = "startSession"
        const val JOIN_PRESENTATION = "joinPresentation"
        const val END_SESSION = "endSession"
        const val SWITCH_CAMERA = "switchCamera"
        const val MUTE_AUDIO = "muteAudio"
        const val PAUSE_VIDEO = "pauseVideo"
        const val SET_AUDIO_MODE = "setAudioMode"
        const val ENTER_PIP_MODE = "enterPIPMode"
        const val EXIT_PIP_MODE = "exitPIPMode"
        const val SWITCH_TO_VIDEO_CALL = "switchToVideoCall"
        const val START_RECORDING = "startRecording"
        const val STOP_RECORDING = "stopRecording"
        const val SUCCESS = "success"
        const val ERROR = "error"
    }

    object Code{
        const val SUCCESS = "SUCCESS"
        const val ERROR = "ERROR"
    }

    object ErrorMessage{
        const val GENERATED_TOKEN_NULL = "Generated token is null"
        const val CALL_TOKEN_NULL = "Call token is null"
        const val EVENT_STREAM_SINK_NULL = "Event stream sink is null"
        const val SETTINGS_METHOD_CALL_ERROR = "Unable to process your request"
        const val CALLS_SDK_NOT_INITIALIZED = "Please initialize the CometChatCalls by calling CometChatCalls.init() method"
        const val ARGUMENT_IS_NULL = "Passed argument is null"
    }

    object SuccessMessage{
        const val INIT = "CometChat SDK initialized successfully"
        const val SUCCESS = "success"
    }

}